modules/custom_operations/user_ie_extensions/sentence_piece/sentence_piece.cpp

```diff
@@ -0,0 +1,186 @@
+
+    evaluate_helper(inputs, m_sparse_indices, m_sparse_values, m_sparse_dense_shape);
+
+    outputs[0].set_shape({ m_sparse_indices.size() / 2, 2 });
+    memcpy(outputs[0].data(), m_sparse_indices.data(), sizeof(int32_t) * m_sparse_indices.size());
+    outputs[1].set_shape({ m_sparse_values.size() });
+    memcpy(outputs[1].data(), m_sparse_values.data(), sizeof(int32_t) * m_sparse_values.size());
+    outputs[2].set_shape({ 2 });
+    memcpy(outputs[2].data(), m_sparse_dense_shape.data(), sizeof(int32_t) * m_sparse_dense_shape.size());
+    return true;
+}
+
+bool SentencepieceTokenizer::has_evaluate() const {
+    return true;
+}
+
+std::shared_ptr<ov::Node> SentencepieceTokenizer::clone_with_new_inputs(const ov::OutputVector& new_args) const {
+    return std::make_shared<SentencepieceTokenizer>(new_args);
+}
+
+OutputVector translate_sentencepiece_op(const ov::frontend::NodeContext& node) {
+    // extract model to configure SentencePieceTokenizer
+    auto sp_model_ov_any = node.get_attribute_as_any("model");
+    FRONT_END_GENERAL_CHECK(sp_model_ov_any.is<std::string>(),
+        "SentencePieceOp configuration model is in incorrect format");
+    auto str_spm_model = sp_model_ov_any.as<std::string>();
+    auto sp_model_const = std::make_shared<Constant>(element::u8, Shape{ str_spm_model.size() }, str_spm_model.data());
+    return { sp_model_const };
+}
+
+OutputVector translate_sentencepiece_tokenizer(const ov::frontend::NodeContext& node) {
+    // this is custom translator that converts a sub-graph with SentencePieceOp, SentencePieceTokenizer,
+    // and RaggedTensorToSparse operation- into a custom operation SentencepieceTokenizerExtensionOp
+    FRONT_END_GENERAL_CHECK(node.get_input_size() > 0, "RaggedTensorToSparse expects at least one input.");
+
+    // check that producers of RaggedTensorToSparse is SentencePieceTokenizer
+    auto sp_tokenize_op = node.get_input(0).get_node_shared_ptr();
+    FRONT_END_GENERAL_CHECK(sp_tokenize_op->get_input_size() > 6,
+        "SentencepieceTokenizeOp expects at least six inputs");
+
+    // prepare inputs that go to custom operation
+    // prepare input 0 - SentencePieceTokenizer configuration model
+    auto sp_model_const = sp_tokenize_op->input_value(0).get_node_shared_ptr();
+
+    // prepare input six inputs
+    auto inputs = sp_tokenize_op->input_value(1);
+    auto nbest_size = sp_tokenize_op->input_value(2);
+    auto alpha = sp_tokenize_op->input_value(3);
+    auto add_bos = sp_tokenize_op->input_value(4);
+    auto add_eos = sp_tokenize_op->input_value(5);
+    auto reverse = sp_tokenize_op->input_value(6);
+
+    OutputVector inputs_vector = OutputVector{ sp_model_const, inputs, nbest_size, alpha, add_bos, add_eos, reverse };
+
+    // Override type of input tensor if this is a Parameter
+    if(auto parameter = std::dynamic_pointer_cast<ov::opset10::Parameter>(inputs.get_node_shared_ptr())) {
+        parameter->set_partial_shape(ov::PartialShape{Dimension()});
```

Comment:
@rkazants, why was the diagnostics message about change of parameter type removed? It warns the user that extension is changing the model signature which is potentially destructive and shouldn't be applied to any model.